### PR TITLE
Remove `status` dropdown for auction form

### PR DESCRIPTION
**Pull request by jessieay**
* Update status via bid box only
* This revealed a bug, which was that we didn't check that a vendor was
  eligible when accepting an auction via the bid box
* This PR moves the "vendor eligible" code to `AcceptAuction`
* Closes https://github.com/18F/micropurchase/issues/1243

### CODE REVIEW

File: app/controllers/admin/auction_acceptances_controller.rb [hunk 1]
```diff
@@ -2,12 +2,18 @@ class Admin::AuctionAcceptancesController < Admin::BaseController
...
     @auction = Auction.find(params[:id])
 
-    AcceptAuction.new(
+    accept_auction = AcceptAuction.new(
       auction: @auction,
       payment_url: winning_bidder.payment_url
-    ).perform
+    )
+
+    if accept_auction.perform
```
Comment (harrisj) [inline app/controllers/admin/auction_acceptances_controller.rb]: ooof, good catch

File: app/services/accept_auction.rb [hunk 2]
```diff
@@ -7,6 +7,17 @@ def initialize(auction:, payment_url: '')
   end
 
   def perform
+    if vendor_ineligible?
+      auction.errors.add(:base, I18n.t('errors.update_auction.vendor_ineligible'))
+      false
+    else
```
Comment (harrisj) [inline app/services/accept_auction.rb]: Oh man, well done

File: app/services/accept_auction.rb [hunk 3]
```diff
@@ -19,7 +30,36 @@ def perform
...
 
-  private
+  def vendor_ineligible?
+    !winning_bidder_is_eligible_to_be_paid?
+  end
+
+  def winning_bidder_is_eligible_to_be_paid?
+    if auction_is_small_business?
+      reckoner = SamAccountReckoner.new(winning_bidder)
+      reckoner.set!
```
Comment (harrisj) [inline app/services/accept_auction.rb]: This is basically a saving throw to see if their SAM status is good before bid right? Does the reckoner have switch from true -> false?

**Comment by harrisj** [comment]
LGTM. Merging

File: app/services/accept_auction.rb (hunk 3 above)
Comment (jessieay) [inline app/services/accept_auction.rb]: this is all happening when an auction is being accepted. I think the assumption we are making is that someone cannot become sam-unverified, but they can go from small business to not small business between the time they bid and when they win.... perhaps we will want to re-visit that assumption at some point

**Pull request merged**